javascript-data-model/main.js

```diff
@@ -0,0 +1,55 @@
+/* exported orderHistory */
+
+var orderHistory = [
+  {
+    name: "JavaScript for impatient programmers",
+    type: "book",
+    by: "Rauschmayer, Dr. Axel",
+    price: 31.55,
+    deliveredDate: 08082020,
+    returnByDate: 09072020,
+    orderNumber: 11439416898772232
+  },
+  {
+    name: "The Timeless Ways of Building",
+    type: "book",
+    by: "Alexander, Christopher",
+    price: 41.33,
+    deliveredDate: 07202020,
+    returnByDate: 08192020,
+    orderNumber: 11399842681280257
+  },
+  {
+    name: "Gamecube Controller Adapter",
+    type: "wii U or PC",
+    by: "Super Smash Bros Switch",
+    price: 15.98,
+    deliveredDate: 07072020,
+    returnByDate: 08052020,
+    orderNumber: 11428755579059409
+  },
+
+  {
+    [
```

Comment:
This array has no property name it is being stored in. Also, all of your other orders in the orderHistory array are just a single object, but this object holds an array. 

> When the data type of the value differs like this between entries, looping over your data becomes a nightmare. Try to make sure that your data model and the types each value holds are the same between all entries. The data itself will vary, but the structure they are represented in should always be the same.